nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-api/src/main/java/org/apache/nifi/authorization/AuthorizeControllerServiceReference.java

```diff
@@ -57,8 +58,12 @@ public static void authorizeControllerServiceReferences(final Map<String, String
                     if (!Objects.equals(currentValue, proposedValue)) {
                         // ensure access to the old service
                         if (currentValue != null) {
-                            final Authorizable currentServiceAuthorizable = lookup.getControllerService(currentValue).getAuthorizable();
-                            currentServiceAuthorizable.authorize(authorizer, RequestAction.READ, user);
+                            try {
+                                final Authorizable currentServiceAuthorizable = lookup.getControllerService(currentValue).getAuthorizable();
+                                currentServiceAuthorizable.authorize(authorizer, RequestAction.READ, user);
+                            } catch (ResourceNotFoundException e) {
+                                // ignore if the resource is not found, if currentValue was previously deleted, it should not stop assignment of proposedValue
```

Comment:
That section of code can still raise an AccessDeniedException, we're just ignoring the RNFE in this specific case.  There are probably other cases where the RNFE would bubble up to the UI, where the exception mapper would handle it properly, but in this case it was preventing the new controller service to be assigned.  I think for 1.0.0, this is the best fix.